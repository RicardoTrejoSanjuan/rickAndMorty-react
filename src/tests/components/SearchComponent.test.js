import React from 'react';
const { shallow } = require("enzyme");
const { default: SearchComponent } = require("../../components/SearchComponent");

describe('Test en <SearchComponent />', () => {

    const setCharacter = (e) => console.log(e);
    
    test('Dede mostrar el componente', () => {
        const wrapper = shallow(<SearchComponent setCharacter={ setCharacter } />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe contener un <div className="card">', () => {
        const wrapper = shallow(<SearchComponent setCharacter={ setCharacter } />);

        expect(wrapper.find('form').exists()).toBe(true);
    });

});