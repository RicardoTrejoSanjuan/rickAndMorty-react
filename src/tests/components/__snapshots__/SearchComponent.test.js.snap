// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <SearchComponent /> Dede mostrar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      key="inputValue"
    >
      <span
        className="text-danger"
      >
        *
      </span>
       Character: 
    </label>
    <input
      className="form-control"
      id="inputValue"
      onChange={[Function]}
      placeholder="Introduce One Character"
      required={true}
      type="text"
      value=""
    />
  </div>
</form>
`;
